streamlit-dynamic-filters/dynamic_filters.py

```diff
@@ -140,33 +143,179 @@ def display_filters(self, location=None, num_columns=0, gap="small"):
             max_value = num_columns
             col_list = st.columns(num_columns, gap=gap)
 
-        for filter_name in st.session_state.filters.keys():
+        for filter_name in st.session_state[self.filters_name].keys():
             filtered_df = self.filter_df(filter_name)
             options = filtered_df[filter_name].unique().tolist()
 
             # Remove selected values that are not in options anymore
-            valid_selections = [v for v in st.session_state.filters[filter_name] if v in options]
-            if valid_selections != st.session_state.filters[filter_name]:
-                st.session_state.filters[filter_name] = valid_selections
+            valid_selections = [v for v in st.session_state[self.filters_name][filter_name] if v in options]
+            if valid_selections != st.session_state[self.filters_name][filter_name]:
+                st.session_state[self.filters_name][filter_name] = valid_selections
                 filters_changed = True
 
             if location == 'sidebar':
                 with st.sidebar:
-                    selected = st.multiselect(f"Select {filter_name}", options, default=st.session_state.filters[filter_name])
+                    selected = st.multiselect(f"Select {filter_name}", options,
+                                              default=st.session_state[self.filters_name][filter_name])
             elif location == 'columns' and num_columns > 0:
                 with col_list[counter - 1]:
-                    selected = st.multiselect(f"Select {filter_name}", options, default=st.session_state.filters[filter_name])
+                    selected = st.multiselect(f"Select {filter_name}", options,
+                                              default=st.session_state[self.filters_name][filter_name])
 
                 # increase counter and reset to 1 if max_value is reached
                 counter += 1
                 counter = counter % (max_value + 1)
                 if counter == 0:
                     counter = 1
             else:
-                selected = st.multiselect(f"Select {filter_name}", options, default=st.session_state.filters[filter_name])
+                selected = st.multiselect(f"Select {filter_name}", options,
+                                          default=st.session_state[self.filters_name][filter_name])
 
-            if selected != st.session_state.filters[filter_name]:
-                st.session_state.filters[filter_name] = selected
+            if selected != st.session_state[self.filters_name][filter_name]:
+                st.session_state[self.filters_name][filter_name] = selected
+                filters_changed = True
+
+        if filters_changed:
+            st.rerun()
+
+    def display_df(self, **kwargs):
+        """Renders the filtered dataframe in the main area."""
+        # Display filtered DataFrame
+        st.dataframe(self.filter_df(), **kwargs)
+
+
+class DynamicFiltersHierarchical(DynamicFilters):
+

```

Comment:
@kzielins could you add a docstring with some explanation of what DynamicFiltersHierarchical Class does. thanks